cypress-io/cypress
# feat(unify): Add "No Specs Found" screen

- Closes https://cypress-io.atlassian.net/browse/UNIFY-600

### User facing changelog

Adds the "No Specs Found" UI, which will be shown when there are no specs found matching the `specPattern` but only when the `specPattern` has been modified from the default.

### Additional details

The main change here is that the CreateSpecPage has been broken up. The top-level template is now a NoSpecsPage that will render different content depending on the situation. The logic for this will come in a [follow up ticket](https://cypress-io.atlassian.net/browse/UNIFY-625), but this PR adds the UI and structure for it. 

### How has the user experience changed?

Previously if no specs were found matching the specPattern, we would always show the "Create Spec"  messages and buttons. Now, if there is a custom spec pattern in use, we show the pattern and suggest people validate the spec pattern. Otherwise, we show the same screen as before.

Screenshots:

If there are no specs found and the default specPattern has **not** been changed, we show the same Create Spec content:

<img width="1333" alt="Screen Shot 2021-11-30 at 9 51 51 AM" src="https://user-images.githubusercontent.com/8340719/144069982-330cf8ea-abb2-44b5-b100-8a9e90db4305.png">

If there are no specs found and the default specPattern **has** been changed, we show the new No Specs Found content:

<img width="1331" alt="Screen Shot 2021-11-30 at 9 51 14 AM" src="https://user-images.githubusercontent.com/8340719/144070214-190d5a0e-0e82-41b2-b230-005028858452.png">

### How to test

1. Using `yarn dev`, open a project in the Cypress app that has no specs, or break the spec pattern in a project and then open it. Visiting the specs list should show the usual "Create New Spec" screen - even with a broken spec pattern, since that logic is not in this PR. 

2. Modify the ref `isUsingDefaultSpecs` in `app/src/pages/Index.vue` to be `false` and refresh the app, you should now see the "No Specs Found" version of the screen.

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
